src/Aspire.Hosting/Extensions/ExecutableResourceBuilderExtensions.cs

```diff
@@ -59,7 +59,21 @@ public static IResourceBuilder<ExecutableResource> AddExecutable(this IDistribut
         return builder.WithManifestPublishingCallback(context => WriteExecutableAsDockerfileResourceAsync(context, builder.Resource));
     }
 
-    private static async Task WriteExecutableAsDockerfileResourceAsync(ManifestPublishingContext context, ExecutableResource executable)
+    /// <summary>
+    /// Adds annotation to <see cref="ExecutableResource" /> to support containerization during deployment.
+    /// The resulting container image is built with the given <paramref name="buildArgs"/> that are provided
+    /// during <c>docker build</c>, as <c>--build-arg</c>.
+    /// </summary>
+    /// <typeparam name="T">Type of executable resource</typeparam>
+    /// <param name="builder">Resource builder</param>
+    /// <param name="buildArgs"></param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<T> PublishAsDockerFile<T>(this IResourceBuilder<T> builder, DockerBuildArg[] buildArgs) where T : ExecutableResource
```

Comment:
Why 2 overloads?

Reply:
Is it preferred to add an optional parameter to the existing method?